packages/merkle-distributor/scripts/2_PublishClaimsForWindow.ts

```diff
@@ -0,0 +1,126 @@
+//       "0xf0987f2e0ec07c35d7cfdd4f5831d8ef2a1829413321611caecbf08eaddf567c",
+//       "0xedd0b39fe99a555994df53dfca180b84437abbd1b808fbadd5e8a70f7f2a40cf"
+//     ]
+//   }... for all claims
+// }
+
+// example execution: ts-node ./scripts/2_PublishClaimsForWindow.ts -i ./proof-files/chain-id-42-reward-window-0-claims-file.json -m 0xAfCd2405298C2FABB2F7fCcEB919B4505A6bdDFC --network kovan_mnemonic
+
+const assert = require("assert");
+import { program } from "commander";
+import fs from "fs";
+
+const { getAbi } = require("@uma/core");
+const { getWeb3, MAX_UINT_VAL } = require("@uma/common");
+const web3 = getWeb3();
+const { toBN } = web3.utils;
+
+import IpfsHelper = require("../src/IpfsHelper");
+import CloudflareKVHelper = require("../src/CloudflareKVHelper");
+
+program
+  .option("-n --network")
+  .requiredOption("-i, --input <path>", "input JSON file location containing a recipients payout")
+  .requiredOption("-m, --merkleDistributorAddress <address>", "address of the merkle distributor contract")
+  .parse(process.argv);
+
+const options = program.opts();
+
+const claimsObject = JSON.parse(fs.readFileSync(options.input, { encoding: "utf8" }));
+
+// We can't easily do runtime verification of JSON file types in typescript. We could use a JSON scheme, but to keep
+// things simple for now we can just double check that some important keys are present within the JSON file.
+if (typeof claimsObject !== "object") throw new Error("Invalid JSON");
+const expectedKeys = ["chainId", "rewardToken", "windowIndex", "totalRewardsDistributed", "windowStart", "claims"];
+expectedKeys.forEach(expectedKey => {
+  if (!Object.keys(claimsObject).includes(expectedKey)) {
+    throw new Error("claims object missing expected key");
+  }
+});
+
+const merkleDistributor = new web3.eth.Contract(getAbi("MerkleDistributor"), options.merkleDistributorAddress);
+
+async function main() {
+  const account = (await web3.eth.getAccounts())[0];
+
+  // Do some basic sanity checks.
+  console.log("0. Running some basic sanity checks on the payout file 🔎");
+  assert(web3.utils.isAddress(options.merkleDistributorAddress), "Invalid merkleDistributorAddress");
+  assert((await merkleDistributor.methods.owner().call()) == account, "Unlocked account does not own the distributor");
+  assert((await web3.eth.net.getId()) == claimsObject.chainId, "The connected network not match your  JSON chainId");
+  assert(web3.utils.isAddress(claimsObject.rewardToken), "Invalid rewardToken" + claimsObject.rewardToken);
+  assert((await merkleDistributor.methods.lastSeededIndex().call()) == claimsObject.windowIndex, "Wrong windowIndex");
+
+  console.log("File passed the checks\n\n1. Adding claims file to IPFS 🛰");
+
+  const ipfsHash = await IpfsHelper.uploadFile(JSON.stringify(claimsObject.claims));
+
+  console.log("Claims file added to IPFS with hash:", ipfsHash, "\n\n2. Pinning claims file to IPFS 📌");
+
+  await IpfsHelper.pinHash(ipfsHash);
```

Comment:
What's the difference between pinning and uploading? Would be great to add to comments too

Reply:
once it is uploaded the pinning makes sure it is persistently accessible on the pinning services.

I have added a comment.